Homework2/DiagonalMatrix/ExtensionMethod.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DiagonalMatrix
+{
+
+    internal static class ExtensionMethod
+    {
+        public static DiagonalMatrix Add(this DiagonalMatrix firstMatrix, DiagonalMatrix secondMatrix)
+        {
+            int[] _firstDiagonal = firstMatrix.diagonalNumbers;
```

Comment:
Let simplify this method.
Pseudo code:

```
int  newSize = firstMatrix.Size > secondMatrix.Size ? firstMatrix.Size : secondMatrix.Size;
int newElements = new int[newSize];
for ( int i = 0; i < newSize; i ++)
{
    newElements[i] = firstMatrix[i,i] + secondMatrix[i,i];   //We do not worry about OutOfRangException, because our indexers return 0 when i is out of size. 
}
return new DiagonalMatrix(newElements);

```